File: app_java/src/main/java/tech/thdev/android_mvp_sample/data/source/image/SampleImageSource.java
package tech.thdev.android_mvp_sample.data.source.image;

import android.content.Context;

import java.util.ArrayList;

import tech.thdev.android_mvp_sample.data.ImageItem;

/**
 * Created by tae-hwan on 1/30/17.
 */

public interface SampleImageSource {

    interface LoadImageCallback {

        void onImageLoaded(ArrayList<ImageItem> list);
    }

    void getImages(Context context, int size, LoadImageCallback loadImageCallback);
}

File: app_kotlin/src/main/java/tech/thdev/app_kotlin/data/source/image/SampleImageSource.kt
package tech.thdev.app_kotlin.data.source.image

import android.content.Context
import tech.thdev.app_kotlin.data.ImageItem
import java.util.*

/**
 * Created by tae-hwan on 1/30/17.
 */
interface SampleImageSource {

    interface LoadImageCallback {

        fun onLoadImages(list: ArrayList<ImageItem>)
    }

    fun getImages(context: Context, size: Int, loadImageCallback: LoadImageCallback?)
}